pkg/vsphere/tasks/waiter.go

```diff
@@ -70,10 +75,18 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 		}
 
 		log.Errorf("task failed: %s", err)
-		if !isTaskInProgress(err) {
+		if !needsRetry(target, err) {
```

Comment:
You are calling `needsFix` again after this; I thought we could get rid of the extra call this way.